R/bin.R

```diff
@@ -1,3 +1,37 @@
+binMedian <- function(data, x, y, group = NULL, panel = NULL, binWidth = NULL, viewport) {
+  data <- data[data[[x]] >= viewport$xMin & data[[x]] <= viewport$xMax,]
+  data$binLabel <- bin(data[[x]], binWidth, viewport)
+
+  byCols <- colnames(data)[colnames(data) %in% c('binLabel', group, panel)]
+  data <- data[, list(value=roundedMedian(get(..y))), by=eval(byCols)]
+
+  data$binStart <- findBinStart(data$binLabel)
+  data$binEnd <- findBinEnd(data$binLabel)
+  data <- data[order(data$binStart),]
+  data$binStart <- as.character(data$binStart)
+
+  data <- collapseByGroup(data, group, panel)
+
+  return(data)
+}
+
+binMean <- function(data, x, y, group = NULL, panel = NULL, binWidth = NULL, viewport) {
```

Comment:
could we distinguish these two binning functions from others that take only a single variable? Maybe like binXMeanY or binMeanXY... idk something to suggest it depends on both x and y.

Reply:
its a good question. i was debating it w myself for a bit, particularly when i got to the proportion option for lineplot and a binProportion already existed. eventually i decided to punt on it for now. i want to think about this group of fxns more, but we need lineplot in the short term